### DISCUSSION TITLE
Other Compatible Models with Cline via OpenRouter

### DISCUSSION THREAD

**Discussion by filnadyir** [cline/cline#726] [Feature Requests]
Hello,
I know that Claude models work on Cline, and I would like to know if there are other compatible models, and if so, which ones. It would be very helpful to have more information about the supported models beyond Claude, especially to understand if there are other alternatives I could use through OpenRouter. Thank you in advance for your help and clarification.

**Comment by damarges** [comment]
Give qwen2.5-coder and qwen2.5-coder-tools a try. especially now with the new 32B models. and don't forget to increase context size (if possible)

**Comment by filnadyir** [comment]
Thank you very much for your response. I tried the Qwen2.5 Coder 32B model on Cline as you mentioned, but unfortunately, this one doesn’t seem to be compatible either. And I can't use it locally, as I don't have the necessary hardware. I’m starting to think that no other model works on Cline at all. 🥺

**Reply by damarges** [reply]
Have you extended your context size to at least 32k?

**Reply by filnadyir** [reply]
> Have you extended your context size to at least 32k?

To be honest, I wouldn't know how to do it. Could you explain it to me?

**Comment by damarges** [comment]
Well thats Not true. There are several YouTube Videos online of people using qwen coder (Tools) with Cline. I did myself yesterday at Home With the quantized 32B model (Q4_K_M) but my Hardware is potato and it was slow as hell. But the results were good

**Reply by filnadyir** [reply]
> Well thats Not true. There are several YouTube Videos online of people using qwen coder (Tools) with Cline. I did myself yesterday at Home With the quantized 32B model (Q4_K_M) but my Hardware is potato and it was slow as hell. But the results were good

Okay, then I can try and see how it goes. Thank you.

**Comment by Bigsy** [comment]
A lot of models work fine, in fact I find qwen coder models work great, the trouble with all these though is limited context length in some configurations, Cline stuffs everything it can in there and while Claude can deal with 200k tokens not many others can, you see that a lot of the qwen models on openrouter are limited to 33k tokens and if you run it via ollam the default is a tiny 4k

**Reply by filnadyir** [reply]
> A lot of models work fine, in fact I find qwen coder models work great, the trouble with all these though is limited context length in some configurations, Cline stuffs everything it can in there and while Claude can deal with 200k tokens not many others can, you see that a lot of the qwen models on openrouter are limited to 33k tokens and if you run it via ollam the default is a tiny 4k

I understand, but my problem is the incompatibility of Cline with other models when using OpenRouter, even though it works perfectly fine with Claude. In this example, for instance, it’s not compatible with Qwen-2.5-coder-32b, and I don’t understand why. In fact, Cline seems to be incompatible with almost all models, or nearly so.
![Code_nVT3M4T65D](https://github.com/user-attachments/assets/570d98fe-d690-44cb-8a19-e54571763aa0)

**Reply by Bigsy** [reply]
Why are you saying its not compatible with Cline? Because some features are in red?

Its just telling you that the model you selected is not vision capable so you can't use the computer use capability.

Were you going to use that feature because its still a very new niche feature probably 1% of people are using. The main use case for this plugin works absolutely fine without those 3 things in red.

Have you simply tried using it? Does it work?

**Reply by filnadyir** [reply]
> Why are you saying its not compatible with Cline? Because some features are in red?
> 
> Its just telling you that the model you selected is not vision capable so you can't use the computer use capability.
> 
> Were you going to use that feature because its still a very new niche feature probably 1% of people are using. The main use case for this plugin works absolutely fine without those 3 things in red.
> 
> Have you simply tried using it? Does it work?

Yes, sorry, you're right, I should have been more precise: actually, some models don't work at all in the sense that they throw errors and can't be used anymore, while with Qwen it is possible to use it, but for some reason, it doesn't work well. I mean, sometimes it doesn't understand what it needs to do and keeps repeating the same thing, or it keeps modifying the code endlessly even when it works fine, or it can't create the file at all, so it can't even start writing the code. At this point, I'll try testing the models locally.
